engine/src/main/java/org/hibernate/validator/internal/properties/javabean/JavaBeanField.java

```diff
@@ -63,6 +66,31 @@ public String getPropertyName() {
 		return getName();
 	}
 
+	@Override
+	public AnnotatedType getAnnotatedType() {
```

Comment:
I like these changes. We don't expose the reflection `Member` from these classes and annotation provider got cleaner. Only question left - should we iterate once more and hide the annotations as well ? I had such idea at first but discarded it till later times. I was thinking about moving all this logic of finding constraints into these `org.hibernate.validator.internal.properties.javabean.*` classes that would make the `AnnotationMetaDataProvider` much smaller. And also "hide" the reflection even more.

Reply:
For now, I don't think we should add the reflection from `AnnotationMetaDataProvider`: it's tied to Java beans anyway. Maybe we will need to do it later but I think it can be kept for another day.